### DISCUSSION TITLE
Support for packages that use subpath imports?

### DISCUSSION THREAD

**Discussion by lewisl9029** [yarnpkg/berry#4081] [Q&A] [answered]
Hi there!

I'm looking to use the [`@jspm/generator`](https://github.com/jspm/generator) package in a yarn 3 project. It installs just fine, but when I try to import it in my app, it throws with the following familiar error, but this time there's a twist:

```
[intake] Error: @jspm/generator tried to access #fetch, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
[intake] 
[intake] Required package: #fetch (via "#fetch/package.json")
```

`#fetch` is not an actual dependency, but rather a lesser known node feature: [subpath imports](https://nodejs.org/dist/latest-v17.x/docs/api/packages.html#subpath-imports)

It's basically a way to conditionally map #-prefixed identifiers to different modules depending on environment.

I'd assume this would apply to any package that uses this feature, not just `@jspm/generator`, but haven't found another package to verify this yet.

Curious if there's a quick and easy way to work around this? And whether an actual fix could be relatively simple? If so, I'd be happy to dive in and try to send in a PR. 

Thanks!

**Comment by arcanis** [comment, marked as answer]
The PnP linker doesn't support `imports` at the moment; we use [`resolve.exports`](https://github.com/lukeed/resolve.exports) to implement this logic, but it doesn't support `imports` yet. A ticket was opened recently:

- https://github.com/lukeed/resolve.exports/issues/14
